app-modules/portal/routes/api.php

```diff
@@ -97,11 +101,40 @@
                 Route::get('/categories', [KnowledgeManagementPortalCategoryController::class, 'index'])
                     ->name('category.index');
 
-                Route::get('/categories/{category}', [KnowledgeManagementPortalCategoryController::class, 'show'])
-                    ->name('category.show');
+                Route::get('/categories/{category:slug}', [KnowledgeManagementPortalCategoryController::class, 'show'])
+                    ->name('category.show')
+                    ->missing(function (Request $request) {
+                        throw_if(! str()->isUuid($request->category), ModelNotFoundException::class);
 
-                Route::get('/categories/{category}/articles/{article}', [KnowledgeManagementPortalArticleController::class, 'show'])
-                    ->name('article.show');
+                        $category = KnowledgeBaseCategory::findOrFail($request->category);
+
+                        return redirect()->route('api.portal.category.show', $category->slug);
+                    });
```

Comment:
Never seen this method used before, nice!

Wondering if we could pass an instance of an invokable class instead of a closure (or convert it to a closure, either `app(Class::class)(...)` or `app(Class::class)->execute(...)`), to keep the logic out of the routes file?

Also applies below

Reply:
Done